### PR TITLE
Fix save state loading with multi-disc PBPs

### PR DESCRIPTION
**Pull request by justinweiss**
I just ran into the problem mentioned in #184, and this is my stab at fixing it.

With this change, I can see the log messages adjusting the PBP image offset when loading savestates, which I didn't before. And I tested it by loading a Disc 2 savestate, and by restarting after loading the savestate and loading a Disc 2 save file. Both seemed to work.

There's only a few relevant lines, but I unfortunately had to move the function below the CDInsert / Eject functions, so it looks like there's more. 

I'm brand new to libretro, and rusty at C++, so I'd appreciate any feedback or suggestions!

### CODE REVIEW

File: libretro.cpp [hunk 1]
```diff
@@ -1993,6 +1940,70 @@ static void CDSelect(void)
...
+      SFEND
+   };
+
+
+   int ret = MDFNSS_StateAction(sm, load, data_only, StateRegs, "MAIN");
+
+   // Call SetDisc() BEFORE we load CDC state, since SetDisc() has emulation side effects.  We might want to clean this up in the future.
+   if(load)
+   {
+      if(CD_IsPBP)
```
Comment (justinweiss) [inline libretro.cpp]: This is the relevant section.

**Comment by rz5** [comment]
@justinweiss Thanks for the commit, I'll merge it since the new behavior is exclusive to PBP files. Don't forget that the bug you referred to has a bounty associated, so once the participants in that thread acknowledge it as fixed, the bounty is yours.

**Comment by Zapeth** [comment]
Well I'm not really familiar with the save state relevant code (specifically regarding the point in time when the according code is executed) so I can't really say much about that, but I guess it should be fine if it resolves the issue.

The only thing I noticed is that [code]CDC->SetDisc()[/code] gets called twice in the code, not sure if thats intended/necessary or not.
If not then you could remove either the CD*Eject() calls like below (and lose the "open/close" log messages)
[code]//CDEject();
CDC->SetDisc(CD_TrayOpen, (CD_SelectedDisc >= 0 && !CD_TrayOpen) ? (*cdifs)[0] : NULL,
         (CD_SelectedDisc >= 0 && !CD_TrayOpen) ? cdifs_scex_ids[0] : NULL);
//CDInsertEject();[/code]
or the CDC->SetDisc() call in between (again, not sure if it needs to be like this so testing would be recommended if its changed).

**Comment by justinweiss** [comment]
Good catch. That wasn't intended -- the right fix would be to remove the `CDC->SetDisc()`. We need `(*cdifs)[0]->Eject(false)` to be called in order to move the PBP file to the right offset, and we also need `CD_TrayOpen` to be true for Eject to have any effect.  I could take a look tonight.

**Pull request merged**